From 13150b8e2b49a9c95eaad831ff69429ff9f66aa4 Mon Sep 17 00:00:00 2001
From: Samuel Martin <s.martin49@gmail.com>
Date: Sat, 30 Aug 2014 11:20:36 +0200
Subject: [PATCH 1/2] cmake: let CMake drives the library type

Do not force CMake to build shared object when BUILD_SHARED_LIBS is off.

Signed-off-by: Samuel Martin <s.martin49@gmail.com>
---
 src/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a0d6179..78946b1 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -34,7 +34,7 @@ SET(libcnary_object_files
 
 SET_SOURCE_FILES_PROPERTIES(${libcnary_object_files} PROPERTIES EXTERNAL_OBJECT true GENERATED true)
 
-ADD_LIBRARY( plist SHARED ${libplist_SRC} ${libcnary_object_files} )
+ADD_LIBRARY( plist ${libplist_SRC} ${libcnary_object_files} )
 TARGET_LINK_LIBRARIES( plist ${LIBXML2_LIBRARIES} )
 SET_TARGET_PROPERTIES( plist PROPERTIES VERSION ${LIBPLIST_LIBVERSION} )
 SET_TARGET_PROPERTIES( plist PROPERTIES SOVERSION ${LIBPLIST_SOVERSION} )
@@ -42,7 +42,7 @@ if(APPLE)
   SET_TARGET_PROPERTIES( plist PROPERTIES INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib")
 endif()
 
-ADD_LIBRARY( plist++ SHARED ${libplist++_SRC} )
+ADD_LIBRARY( plist++ ${libplist++_SRC} )
 TARGET_LINK_LIBRARIES( plist++ plist )
 SET_TARGET_PROPERTIES( plist++ PROPERTIES VERSION ${LIBPLIST_LIBVERSION} )
 SET_TARGET_PROPERTIES( plist++ PROPERTIES SOVERSION ${LIBPLIST_SOVERSION} )
-- 
2.1.0

